src/core/gui/PageView.cpp

```diff
@@ -199,6 +199,12 @@ void XojPageView::endText() {
         }
     }
 
+    if (removeTextEditor) {
+        deleteTextEditor();
+    }
```

Comment:
To me, this doesn't seem to make sense in isolation. If `removeTextEditor` is false, the text editor will be left in an invalid state, since the text element will no longer be in the document anymore.